### PR TITLE
[incubator-kie-7100] fix(drools): evaluate alpha constraints during isAllowedCachedLeft in CombinedConstraint

### LINKED ISSUE #7100: Drools Sequential: Mixed alpha + beta expressions in one step → ClassCastException
Reported in https://github.com/apache/incubator-kie/pull/6909#issuecomment-5616000126

### Mixed alpha + beta expressions in one step → `ClassCastException`

When a single step has both an alpha expression and a beta expression referencing the anchor:

```java
sequence(
        pattern(toy)
                .expr("alpha", t -> t.getName().startsWith("a"))
                .expr("beta", person, (t, p) -> t.getName().equals(p.getName()))
)
```

`PhreakNodeFactory` wraps the constraints in `CombinedAlphaConstraint`, whose `getRequiredDeclarations()` always returns an empty array. `DynamicFilter` then evaluates the whole thing as an alpha constraint and never supplies the anchor tuple to the beta part, which causes a `ClassCastException`.

A single beta expression works fine (existing tests cover that). The issue is only with mixed alpha+beta in the same step.

If supporting this combination is out of scope for now, it would be good to reject it at build time with a descriptive error.

**Relevant code:**
- `CombinedAlphaConstraint.java:54`
- `PhreakNodeFactory.java:184`
- `DynamicFilter.java:59`

### PR DESCRIPTION
**Pull request by Rikkola**
## Summary

When a sequence step contains both **alpha** (intra-fact) and **beta** (cross-fact / anchor-bound) constraints, `DynamicFilter` takes the beta evaluation path and delegates to `CombinedConstraint#isAllowedCachedLeft`.

Previously, `isAllowedCachedLeft` iterated over child constraints and only evaluated those implementing `BetaConstraint`, completely skipping pure `AlphaNodeFieldConstraint` children. This caused facts that satisfied beta constraints but violated alpha constraints to incorrectly pass evaluation and trigger matches.

## Changes

- Renamed `CombinedAlphaConstraint` → `CombinedConstraint` to reflect its support for mixed alpha and beta constraints.
- Evaluating non-`BetaConstraint` children via `isAllowed(FactHandle, ValueResolver)` inside `isAllowedCachedLeft`.
- Added comprehensive unit and edge-case tests for mixed and multiple alpha constraints.

## Issue

Closes: https://github.com/apache/incubator-kie/issues/7100

### CODE REVIEW

File: drools-base/src/main/java/org/drools/base/rule/constraint/CombinedConstraint.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                }
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public boolean isAllowedCachedRight(BaseTuple tuple, CombinedContextEntry context) {
+        throw new UnsupportedOperationException(
+                "CombinedAlphaConstraint does not support right-cached evaluation");
```
[the remark this answers is not in the document]
Comment (Rikkola) [reply drools-base/src/main/java/org/drools/base/rule/constraint/CombinedConstraint.java]: Fixed

File: drools-base/src/test/java/org/drools/base/rule/constraint/CombinedConstraintTest.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * permissions and limitations under the License.
+ */
```
[the remark this answers is not in the document]
Comment (Rikkola) [reply drools-base/src/test/java/org/drools/base/rule/constraint/CombinedConstraintTest.java:18]: Done

**Review by tkobayas — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Review by yesamer — APPROVED** [review]
